backend/wmg/data/validation/validation.py

```diff
@@ -115,6 +120,67 @@ def validate_cube_species(self):
 
             # todo check/log cell type per species
 
+    def validate_tissue_rollup_cell_count(self):
+        """
+        Validates that tissues rolled up in the axis "tissue_ontology_term_id" was correctly done from
+        attribute "tissue_original_ontology_term_id".
+
+        The way this validation works is by creating two cell count tables:
+            1. Create cell count tables for Lung using "tissue_ontology_term_id"
+            2. Create cell count tables for all Lung parts using "tissue_original_ontology_term_id"
+
+        The two should be identical
+        """
+
+        with tiledb.open(self.cell_count_path, "r") as cell_count_cube:
+            human_ontology_id = fixtures.validation_species_ontologies["human"]
+            all_lung_tissues = fixtures.validation_all_lung_tissues
+            lung_high_level_tissue = fixtures.validation_lung_high_level
+
+            original_tissues_cell_count = cell_count_cube.df[:, all_lung_tissues, human_ontology_id].n_cells.sum()
+            rollup_tissues_cell_count = cell_count_cube.df[lung_high_level_tissue, :, human_ontology_id].n_cells.sum()
+
+            if original_tissues_cell_count != rollup_tissues_cell_count:
+                logger.error(
```

Comment:
Question: is this validator meant to be run manually by users or as part of the pipeline? I recall the latter being the original idea but I'm not sure we ended up implementing it. Regardless, I think it would be a good idea to make sure these methods either return a Boolean (valid/invalid) or a list of errors, kinda like the cellxgene-schema validator.

Reply:
It runs automatically as part of the pipeline

https://github.com/chanzuckerberg/single-cell-data-portal/blob/9a8d94c20debc40cbbcdd63e79c775e6e8543fa1/backend/corpus_asset_pipelines/summary_cubes/__init__.py#L14-L30